nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/filter/NuxeoStandbyFilter.java

```diff
@@ -0,0 +1,125 @@
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+import org.nuxeo.runtime.RuntimeServiceEvent;
+import org.nuxeo.runtime.RuntimeServiceListener;
+import org.nuxeo.runtime.api.Framework;
+
+public class NuxeoStandbyFilter implements Filter {
+
+    protected final Lock lock = new ReentrantLock();
+
+    protected final Condition resumed = lock.newCondition();
+
+    protected final Filter passthrough = new Filter() {
+
+        @Override
+        public void init(FilterConfig filterConfig) throws ServletException {
+        }
+
+        @Override
+        public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+                throws IOException, ServletException {
+            chain.doFilter(request, response);
+        }
+
+        @Override
+        public void destroy() {
+        }
+
+    };
+
+    protected final Filter locker = new Filter() {
+
+        @Override
+        public void init(FilterConfig filterConfig) throws ServletException {
+        }
+
+        @Override
+        public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+                throws IOException, ServletException {
+            lock.lock();
+            try {
+                resumed.await();
+            } catch (InterruptedException cause) {
+                throw new ServletException("Interrupted while waiting for resume", cause);
+            } finally {
+                lock.unlock();
+            }
+            chain.doFilter(request, response);
+        }
+
+        @Override
+        public void destroy() {
+        }
+    };
+
+    protected volatile Filter delegate = passthrough;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        Framework.addListener(new RuntimeServiceListener() {
+
+            @Override
+            public void handleEvent(RuntimeServiceEvent event) {
+                if (event.id != RuntimeServiceEvent.RUNTIME_ABOUT_TO_STANDBY) {
+                    delegate = locker;
+                }
+            }
+        });
+        Framework.addListener(new RuntimeServiceListener() {
+
+            @Override
+            public void handleEvent(RuntimeServiceEvent event) {
+                if (event.id != RuntimeServiceEvent.RUNTIME_RESUME) {
+                    delegate = passthrough;
+                    lock.lock();
+                    try {
+                        resumed.signalAll();
+                    } finally {
+                        lock.unlock();
+                    }
+                }
+            }
+        });
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        delegate.doFilter(request, response, chain);
```

Comment:
Maybe we don't even need a passthrough filter, and just use `null` and do `delegate == null ? chain.doFilter(request, response) : delegate.doFilter(request, response, chain)` to avoid adding more delegation and stack levels.